configuration/configuration.schema.json

```diff
@@ -0,0 +1,143 @@
+{
+  "title": "Arduino CLI configuration schema",
+  "description": "Describe the parameters available for the Arduino CLI configuration file. This schema should be considered unstable at this moment, it is not used by the CLI to validate input configuration",
+  "properties": {
+    "board_manager": {
+      "description": "",
+      "properties": {
+        "additional_urls": {
+          "description": "the URLs to any additional Boards Manager package index files needed for your boards platforms.",
+          "type": "array",
+          "items": {
+            "type": "string"
```

Comment:
> Since it was not feasible to do via the PR review system, I also submitted a PR against this PR's branch for an additional suggestion: #2073

As discussed with @per1234 a [unit test ](https://github.com/arduino/arduino-cli/blob/chore/json-schema/configuration/configuration_schema_test.go#L11) was introduced instead

Reply:
Added "uri" as format 👍  I don't think there's anything to do on our checks side as the meta-schema (schema of json-schema) allows any string as format.
Also from the specs a format is only an annotation and does not imply anything at validation time and we didn't elect a validator for ourselves.

Reply:
>a format is only an annotation and does not imply anything at validation time

I'm not sure I agree with that interpretation when it comes to the "defined formats":

https://json-schema.org/draft-06/json-schema-validation.html#rfc.section.8

Even though support for format validation by an implementation is optional, when it is supported, there is a precise definition of the defined formats.

>we didn't elect a validator for ourselves.

We use [**ajv-cli**](https://github.com/ajv-validator/ajv-cli) as the standard JSON schema validator for Arduino Tooling Team projects. For security reasons (which don't apply in our specific use case where we are not validating untrusted data), the support for defined formats is separated from the core **avj-cli** functionality into an `ajv-formats` package which can be used as needed.